messages/teleporter/message_manager.go

```diff
@@ -105,6 +105,32 @@ func (m *messageManager) GetDestinationBlockchainID(unsignedMessage *warp.Unsign
 	return teleporterMessage.DestinationBlockchainID, nil
 }
 
+func (m *messageManager) GetOriginSenderAddress(unsignedMessage *warp.UnsignedMessage) (common.Address, error) {
+	teleporterMessage, err := m.parseTeleporterMessage(unsignedMessage)
+	if err != nil {
+		m.logger.Error(
+			"Failed get teleporter message.",
+			zap.String("warpMessageID", unsignedMessage.ID().String()),
+		)
+		return common.Address{}, err
+	}
+	m.teleporterMessageCache.Put(unsignedMessage.ID(), teleporterMessage)
```

Comment:
Can we move this repeated section to a helper somehow? Looks like we have it 4 or more times in this file now, so even though it doesn't save many lines it will be less error prone as far as the log message and adding the parsed message to the cache.

Reply:
Sure thing. The most straightforward change would be to update the cache in the call to `parseTeleporterMessage` if it's not already there.